driver-sync/src/test/functional/com/mongodb/client/RetryableWritesProseTest.java

```diff
@@ -0,0 +1,205 @@
+
+        // Using the client under test, insert a document and observe a successful write result. The test MUST assert
+        // that the insert command fails once against the stepped down node and is successfully retried on the newly
+        // elected primary (after SDAM discovers the topology change). The test MAY use APM or another means to observe
+        // both attempts.
+        try {
+            System.out.println("--- insert one doc, should fail...");
+            collection.insertOne(new Document("_id", 2).append("x", 22));
+            fail("Exception should have been raised on insert");
+        } catch (MongoException ex) {
+            System.out.println("Expected exception raised: " + ex.getMessage());
+        }
+
+        try {
+            System.out.println("--- insert one doc, should pass...");
+            collection.insertOne(new Document("_id", 2).append("x", 22));
+            System.out.println("--- DONE inserting one doc");
+        } catch (MongoException ex) {
+            fail("Exception should not have been raised on insert: " + ex.getMessage());
+        }
+    }
+
+    private boolean canRunTests() {
+        return serverVersionAtLeast(3, 6) && isDiscoverableReplicaSet();
+    }
+
+    private void setUpClientUnderTest() {
+        MongoClientSettings settings = getMongoClientSettingsBuilder()
+                .retryWrites(true)
+                .applyToServerSettings(new Block<ServerSettings.Builder>() {
+                    @Override
+                    public void apply(final ServerSettings.Builder builder) {
+                        builder.heartbeatFrequency(60000, TimeUnit.MILLISECONDS);
+                    }
+                })
+                .build();
+
+        clientUnderTest = MongoClients.create(settings);
+        System.out.println("setUpClientUnderTest: success");
+    }
+
+    private void setUpFailPointClient() {
+        failPointClient = MongoClients.create(getMongoClientSettingsBuilder().build());
+        System.out.println("setUpFailPointClient: success");
+    }
+
+    private void setUpStepDownClient() {
+        stepDownClient = MongoClients.create(getMongoClientSettingsBuilder().build());
+        System.out.println("setUpStepDownClient: success");
+    }
+
+    private void activateFailPoint() {
+        MongoDatabase adminDB = failPointClient.getDatabase("admin");
+        Document command =
+                new Document("configureFailPoint", "stepdownHangBeforePerformingPostMemberStateUpdateActions")
+                        .append("mode", "alwaysOn");
+        System.out.println("--- activating fail point...");
+        adminDB.runCommand(command);
+        System.out.println("--- DONE activating fail point");
+    }
+
+    private MongoDatabase stepDownPrimary() {
+        final MongoDatabase stepDownDB = stepDownClient.getDatabase("admin");
+        System.out.println("--- stepping down primary...");
+        Thread t = new Thread(new Runnable() {
```

Comment:
Running `replSetStepDown` hangs as expected, thus running the command in a separate thread.